Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -345,7 +345,11 @@ String CanvasRenderingContext2DBase::State::fontString() const
 
     auto italic = font.italic() ? "italic "_s : ""_s;
     auto smallCaps = font.variantCaps() == FontVariantCaps::Small ? "small-caps "_s : ""_s;
-    serializedFont.append(italic, smallCaps, font.computedSize(), "px"_s);
+    serializedFont.append(italic, smallCaps);
+    auto weight = static_cast<int>(font.weight());
+    if (weight != 400)
+        serializedFont.append(weight, " "_s);
+    serializedFont.append(font.computedSize(), "px"_s);
```

Comment:
If I have this script:
```
        ctx.font = "700 48px serif";
        ctx.strokeText("Hello world", 50, 100);
        alert(ctx.font);
```
Should I get `bold 48px serif`? Or should I get `700 48px serif`?

Reply:
That was a good point, with my latest patch we match Chrome behaviour, i.e. 700 gets serialised as bold, and bold remains bold. For completeness, Firefox keeps the value as-is. i.e. 700 -> 700 and bold  -> bold in serialisation.